docs/source/corev_hw_loop.rst

```diff
@@ -43,76 +43,84 @@ Hardware Loop constraints
 
 The HWLoop constraints are:
 
--  Start and End address of an HWLoop must be word aligned
+-  Start and End address of an HWLoop must be word aligned.
+
+-  End address of an HWLoop must point to the instruction just after the last one of the loop body.
 
 -  HWLoop body must contain at least 3 instructions.
-   An illegal exception is raised otherwise.
+   An illegal instruction exception is raised otherwise.
```

Comment:
when does this happen? when setting the END of the HW Loop? We should specify this in my opinion, as well as specify the ordering of settings, because if you check that the HW loop END instruction is greater or equal to START + 12 (or +16 actually as you point to the next after the end)- when setting the END of the loop, then START must be programmed before, otherwise, if you check when fetching the first time the first instruction of the HW Loop, then ordering does not matter.

Reply:
Quite complex ordering constraint.
A Loop Enable bit could solve the ordering problem when programming HW loops CSRs.
General answer could be what is stated in last conversation/comment.

Reply:
sorry what is the last conversation/comment?

Anyway, for me it is fine any kind of solution, e.g. the easiest would be when fetching the first instruction of the HW Loop (start), but we need to say this

Reply:
Do nothing in HW for that ...